script/tool/test/readme_check_command_test.dart

```diff
@@ -275,4 +275,135 @@ A very useful plugin.
       );
     });
   });
+
+  group('code blocks', () {
+    test('fails on missing info string', () async {
+      final Directory packageDir = createFakePackage('a_package', packagesDir);
+
+      packageDir.childFile('README.md').writeAsStringSync('''
+Example:
+
+```
+void main() {
+  // ...
+}
+```
+''');
+
+      Error? commandError;
+      final List<String> output = await runCapturingPrint(
+          runner, <String>['readme-check'], errorHandler: (Error e) {
+        commandError = e;
+      });
+
+      expect(commandError, isA<ToolExit>());
+      expect(
+        output,
+        containsAllInOrder(<Matcher>[
+          contains('Code block at line 3 is missing a language identifier.'),
+          contains('Missing language identifier for code block'),
+        ]),
+      );
+    });
+
+    test('allows unknown info strings', () async {
+      final Directory packageDir = createFakePackage('a_package', packagesDir);
+
+      packageDir.childFile('README.md').writeAsStringSync('''
+Example:
+
+```someunknowninfotag
+A B C
+```
+''');
+
+      final List<String> output = await runCapturingPrint(runner, <String>[
+        'readme-check',
+      ]);
+
+      expect(
+        output,
+        containsAll(<Matcher>[
+          contains('Running for a_package...'),
+          contains('No issues found!'),
+        ]),
+      );
+    });
+
+    test('allows space around info strings', () async {
+      final Directory packageDir = createFakePackage('a_package', packagesDir);
+
+      packageDir.childFile('README.md').writeAsStringSync('''
+Example:
+
+```  dart
+A B C
+```
+''');
+
+      final List<String> output = await runCapturingPrint(runner, <String>[
+        'readme-check',
+      ]);
+
+      expect(
+        output,
+        containsAll(<Matcher>[
+          contains('Running for a_package...'),
+          contains('No issues found!'),
+        ]),
+      );
+    });
+
+    test('passes when excerpt requirement is met', () async {
+      final Directory packageDir = createFakePackage('a_package', packagesDir);
+
+      packageDir.childFile('README.md').writeAsStringSync('''
+Example:
+
+<?code-excerpt "main.dart (SomeSection)"?>
+```dart
+A B C
+```
```

Comment:
Is this what the output of the file will look like after running code-excerpt? Adding the code block below the code-excerpt tag?

Reply:
Yep, you put it above a code block (which can be empty to start) and the script will fill it with the extracted code. You can take a look at the camera README for a real example.

Reply:
I think I've seen it in the Flutter docs site, and it was not super evident how to update the `.md` snippets (or the link to actual code somewhere else).

I think I ended up doing a find and replace first, then learning about a script being available somewhere, and *eventually* managing to run it a while later; here are David's Misadventures With Code-Excerpt As A Complete Noob:

* https://github.com/flutter/website/pull/6933

Reply:
I added https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#readme-code

Hopefully it's reasonably clear; feedback welcome!